islands/ItemsList.tsx

```diff
@@ -0,0 +1,163 @@
+// Copyright 2023 the Deno authors. All rights reserved. MIT license.
+import { useSignal } from "@preact/signals";
+import { useEffect } from "preact/hooks";
+import type { Item } from "@/utils/db.ts";
+import { LINK_STYLES } from "@/utils/constants.ts";
+import IconInfo from "tabler_icons_tsx/info-circle.tsx";
+import UserPostedAt from "@/components/UserPostedAt.tsx";
+
+async function fetchItems(cursor: string) {
+  let url = "/api/items";
+  if (cursor !== "") url += "?cursor=" + cursor;
+  const resp = await fetch(url);
+  if (!resp.ok) throw new Error(`Request failed: GET ${url}`);
+  return await resp.json() as { items: Item[]; cursor: string };
+}
+
+async function fetchVotedItems() {
+  const url = "/api/me/votes";
+  const resp = await fetch(url);
+  if (!resp.ok) throw new Error(`Request failed: GET ${url}`);
+  return await resp.json() as Item[];
+}
+
+function EmptyItemsList() {
+  return (
+    <>
+      <div class="flex flex-col justify-center items-center gap-2">
+        <div class="flex flex-col items-center gap-2 pt-16">
+          <IconInfo class="w-10 h-10 text-gray-400 dark:text-gray-600" />
+          <p class="text-center font-medium">No items found</p>
+        </div>
+
+        <a
+          href="/submit"
+          class="inline-flex items-center justify-center gap-2 rounded-md px-3 py-2 text-primary hover:underline"
+        >
+          Submit your project
+        </a>
+      </div>
+    </>
+  );
+}
+
+interface ItemSummaryProps {
+  item: Item;
+  isVoted: boolean;
+}
+
+function ItemSummary(props: ItemSummaryProps) {
+  const isVoted = useSignal(props.isVoted);
+  const score = useSignal(props.item.score);
+  const url = `/api/items/${props.item.id}/vote`;
+
```

Comment:
Locally, this is what I get when I do the following:
```ts
console.log(props.isVoted); // true/false
console.log(isVoted.value); // undefined
```
This may be a bug. Any ideas, @marvinhagemeister or @kt3k?

Reply:
Somehow using `useComputed()` fixed this 🤷🏾‍♂️